### PR TITLE
Clarify PR caches

### PR DESCRIPTION
**Pull request by plaindocs**
Fix for https://github.com/travis-ci/docs-travis-ci-com/issues/622

(a couple of other tweaks while I was there, which should really have been in a different commit)

### CODE REVIEW

File: user/caching.md [hunk 1]
```diff
@@ -164,40 +164,47 @@ Large files that are quick to install but slow to download do not benefit from c
...
 
-### Fetching and storing caches
+## Fetching and storing caches
 
-* Travis CI fetches the cache for every build, including feature branches and pull requests.
+* Travis CI fetches the cache for every build, including branches and pull requests.
 * There is one cache per branch and language version/ compiler version/ JDK version/  Gemfile location/ etc.
-* Pull requests use the cache of the target of the pull request.
-* If a branch does not have its own cache yet, it uses the master branch cache (unless it is a pull request, see above).
+* If a branch does not have its own cache yet, it uses the master branch cache.
```
Comment (MariadeAnton) [inline user/caching.md]: ah! This one is also the default branch cache, which usually is master

Comment (plaindocs) [inline user/caching.md]: Thanks for the catch!

File: user/caching.md [hunk 2]
```diff
@@ -168,7 +168,7 @@ Large files that are quick to install but slow to download do not benefit from c
 
 * Travis CI fetches the cache for every build, including branches and pull requests.
 * There is one cache per branch and language version/ compiler version/ JDK version/  Gemfile location/ etc.
-* If a branch does not have its own cache yet, it uses the master branch cache.
+* If a branch does not have its own cache, Travis CI fetches the repository default branch cache.
```
Comment (MariadeAnton) [inline user/caching.md]: Sorry for the confusion, @plaindocs - I just double checked and it's master the one we use at the moment for pushes:

https://github.com/travis-ci/travis-build/blob/master/lib/travis/build/script/shared/directory_cache/base.rb#L135

**Comment by MariadeAnton** [comment]
Thank you @plaindocs!

**Pull request merged**